### DISCUSSION TITLE
Strings with colon

### DISCUSSION THREAD

**Discussion by MocioF** [GlotPress/GlotPress-WP#1777] [General]
In translate.wordpress.org importing  a .json with a partial translation of a plugin, if a string in file contains colon (:) character both in original strings and in translation the import fails.

**Comment by pedro-mendonca** [comment]
Hi @MocioF,

I've tested both in translate.wordpress.org and in a local GlotPress.
Imported translations successfully in w.org, and in the local GlotPress could import both originals and translations, successfully.

Here is the JSON example file I've used:
```
{
  "Block: Paragraph": [
    "Bloco: Parágrafo"
  ],
  "Another string to translate.": [
    "Outra string para traduzir."
  ]
}
```

Could you show here the JSON data that is failing to import please?

**Comment by MocioF** [comment]
Hi @pedro-mendonca,
maybe it is a different issue.
This is the file.
[campi-moduli-italiani-it_IT-41c81762c32fb10f3237f4fba31a6376.json](https://github.com/GlotPress/GlotPress/files/14054525/campi-moduli-italiani-it_IT-41c81762c32fb10f3237f4fba31a6376.json)
I'm trying to import translations here: https://translate.wordpress.org/projects/wp-plugins/campi-moduli-italiani/dev/it/default/
but it fails with "0 translations were added".
Thanks for your help

**Comment by pedro-mendonca** [comment]
Hi, 
The file you have has isn't plain JSON, is `Jed 1.x (.json)` because has the below format:
```
{
  "translation-revision-date": "2024-01-25 01:17+0100",
  "generator": "WP-CLI/2.8.1",
  "source": "admin/js/scripts.js",
  "domain": "messages",
  "locale_data": {
    "messages": {
      "": {
        "domain": "messages",
        "lang": "it",
        "plural-forms": "nplurals=2; plural=n != 1;"
      },
      "Confirm filter deletion": [
        "Confermare l'eliminazione del filtro"
      ],
      "Do you really want to delete the filter: ": [
        "Vuoi davvero eliminare il filtro: "
      ]
    }
  }
}
```

If you select the correct format on importing, it should import correctly, I've tested locally with success.

There seems to be an issue related autodetect JED format. The autodetect should be able to distinguish between plain JSON and JED formated JSON.

**Comment by MocioF** [comment]
Thanks,
I imported another jed file, and autodetect worked.
With this one, the import works when declaring JED as format.
So, no problem with semicolon, but only in autodetect feature.
